ios/Trifle/Sources/Digital Signature/Secure Enclave/SecureEnclaveDigitalSignatureKeyManager.swift

```diff
@@ -72,32 +73,53 @@ public class SecureEnclaveDigitalSignatureKeyManager
 
     // MARK: - Internal Methods (DigitalSignatureKeyManager)
 
-    internal func signingKey() throws -> SecureEnclaveSigningKey {
+    internal func createOrLoadSigningKey() throws -> SecureEnclaveSigningKey {
         let signingAlgorithm = Self.keyInfo.signingAlgorithm.attrs
 
-        guard try keyExists() else {
+        if try keyExists() {
+            return try loadSigningKey()
+        } else {
             return try SecureEnclaveSigningKey(generateKeypair(), signingAlgorithm)
         }
-        
-        var keyRef: CFTypeRef?
-        let preparedQuery = SecureEnclaveKeychainQueries.getQuery(with: tag, returnRef: true)
-        SecItemCopyMatching(preparedQuery, &keyRef)
-        
-        return try SecureEnclaveSigningKey(keyRef as! SecKey, signingAlgorithm)
     }
     
+    // MARK: - Internal Methods (DigitalSignatureKeyManager)
+
     internal func verifyingKey() throws -> SecureEnclaveVerifyingKey {
-        let signingKey = try signingKey()
+        // load key if it exists, else throw
+        let signingKey = try loadKey()
 
         guard let publicKey = SecKeyCopyPublicKey(signingKey.privateKey) else {
             throw CryptographicKeyError.unavailablePublicKey
         }
 
         return try SecureEnclaveVerifyingKey(publicKey, Self.keyInfo.signingAlgorithm.attrs)
     }
-    
-    // MARK: -
-    
+
+    // MARK: - Private Methods
+
+    private func loadKey() throws -> SecureEnclaveSigningKey {
```

Comment:
maybe rename this to `fetchSigningKey` or `resolveSigningKey` just to not overload the term load?